### PR TITLE
Creating Fact Modules Uptake

### PR DESCRIPTION
**Pull request by Marymary-dev**
https://thepalladiumgroup.atlassian.net/browse/KHP3-4061

### CODE REVIEW

File: Scripts/NDWH/C&T FACT TABLES/load_FactModulesUptake.sql [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                       County),
+         Htsml
+         As (Select Sdp,
+                    Sdp_agency,
+                    County,
+                    Count (Distinct Sitecode) HTSML
+             From   Ods.Dbo.Hts_eligibilityextract Htsml
+                    Left Join Ods.Dbo.All_emrsites Emr
+                           On Emr.Mfl_code = Htsml.Sitecode
+             Where  Datediff(Month, Visitdate, Eomonth( Dateadd(Mm, -1, Getdate())))  <= 12
```
Comment (nobert-mumo) [inline Scripts/NDWH/C&T FACT TABLES/load_FactModulesUptake.sql:114]: @Marymary-dev for this should we include a check for RiskCategorization to identify ML ?

**Comment by nobert-mumo** [comment]
@Marymary-dev  the logic looks good to me. Only had a few questions/comments:
- I see the fact is an aggregate at partner level. Any reservations on having it a facility level?
- If we go facility level would it make sense to add modules uptake to the EMR Line list instead of creating a new table?

**Comment by Marymary-dev** [comment]
> @Marymary-dev the logic looks good to me. Only had a few questions/comments:
> 
> * I see the fact is an aggregate at partner level. Any reservations on having it a facility level?
> * If we go facility level would it make sense to add modules uptake to the EMR Line list instead of creating a new table?

@nobert-mumo  the fact is at partner and county level. The uptake cannot be at facility level since we would need to compute the uptake for the different program areas.

File: Scripts/NDWH/C&T FACT TABLES/load_FactModulesUptake.sql (hunk 1 above)
Comment (Marymary-dev) [reply Scripts/NDWH/C&T FACT TABLES/load_FactModulesUptake.sql]: @nobert-mumo  this has been included

File: Scripts/NDWH/C&T FACT TABLES/load_FactModulesUptake.sql [hunk 2]
```diff
@@ -0,0 +1,180 @@
+If Object_id(N'[NDWH].[dbo].[FactModulesuptake]', N'U') Is Not Null
+  Drop Table [Ndwh].[Dbo].[FactModulesuptake];
+
+Begin
+    With Sites
+         As (Select 
+         distinct  Mfl_code ,
+                   Sdp,
```
Comment (nobert-mumo) [inline Scripts/NDWH/C&T FACT TABLES/load_FactModulesUptake.sql:8]: @Marymary-dev for the CTEs maybe we can omit the sdp, sdp_Agency, county and sub county to make the code smaller. Then we can get these columns back in the main select.

**Review by nobert-mumo — CHANGES_REQUESTED** [review]
@Marymary-dev for the CTEs maybe we can omit the sdp, sdp_Agency, county and sub county to make the code smaller. Then we can get these columns back in the main select.

**Comment by Marymary-dev** [comment]
> @Marymary-dev for the CTEs maybe we can omit the sdp, sdp_Agency, county and sub county to make the code smaller. Then we can get these columns back in the main select.

@nobert-mumo  this has been cleaned

**Review by nobert-mumo — APPROVED** [review]
LGTM @Marymary-dev, I guess the reporting table will be in another PR..

**Comment by Marymary-dev** [comment]
Sure, look out for line list Module uptake in another PR

**Pull request merged**